scripts/bulk_set_requester_pays.sh

```diff
@@ -0,0 +1,129 @@
+fi
+
+# check for any missing values
+missing=false
+while IFS="," read -r project bucket toggle_type; do
+  if [[ $project == "" ]]
+  then
+    missing=true
+  elif [[ $bucket == "" ]]
+  then
+    missing=true
+  elif [[ $toggle_type == "" ]]
+  then
+    missing=true
+  fi
+done < $CSVIN
+if ( $missing ) ; then
+  echo "ERROR: Missing values in input CSV file."
+  echo "Input CSV file should contain the following columns and corresponding values: project, bucket, toggle_type"
+  exit 0
+else
+  echo "Valid input file."
+fi
+
+# 2. grant the "RequesterPaysToggler" role to the service account for all projects in the list
+# you need to be logged in with your firecloud account
+echo "Enabling permissions for ${SERVICE_ACCOUNT} to set Requester Pays"
+while IFS="," read -r PROJECT_ID; do
+  if gcloud projects add-iam-policy-binding $PROJECT_ID --member=$MEMBER --role=$ROLE --no-user-output-enabled; then
+    echo "Access granted to $PROJECT_ID"
+  fi
+done < <(cut -d "," -f1 $CSVIN | tail -n +2)
+
+# we need to use a service account to set requester pays since the command runs in a loop and we can't authenticate with our regular firecloud account
+# get key
+docker run -v $HOME:/root --rm -it broadinstitute/dsde-toolbox vault read --format=json secret/dsde/prod/common/requester-pays.json | jq .data > rp-key.json
+# authenticate as service account
+gcloud auth activate-service-account ${SERVICE_ACCOUNT} --project="broad-dsde-prod" --key-file=rp-key.json
+rm rp-key.json
+
+# wait for IAM changes to propagate
+echo ""
+echo "Gatorcounting for $SLEEP_SEC seconds while IAM change goes into effect"
+echo ""
+echo "NOTE: if you get an error message saying AccessDeniedException: 403, don't worry, just wait for the next retry."
+echo ""
+echo "waiting $SLEEP_SEC seconds before first attempt"
+echo ""
+sleep $SLEEP_SEC
+
+
+# 3. loop through list of workspace buckets and toggle requester pays on/off
+# for each bucket, retry up to 6 times
+while IFS="," read -r BUCKET TOGGLE; do
+  TOGGLE_TYPE=$(echo $TOGGLE | tr -d '\r')
+  echo "Setting Requester Pays *$TOGGLE_TYPE* for $BUCKET"
+  # when disabling requester pays, a project to bill must be included in the request
+  # we use 'broad-dsde-prod' as the project to bill since all firecloud accounts should already have access to it
+  COUNTER=0
+  if [[ $TOGGLE_TYPE == off ]]; then
+    PROJECT_TO_BILL="-u broad-dsde-prod"
+    else
+      PROJECT_TO_BILL=""
+  fi
+  if [[ $BUCKET == "gs://"* ]]; then
+      BUCKET_PATH=$BUCKET
+    else
+      BUCKET_PATH="gs://$BUCKET"
+  fi
+  while ! gsutil ${PROJECT_TO_BILL} requesterpays set ${TOGGLE_TYPE} ${BUCKET_PATH}
+    do
+      if (( $COUNTER > 5 )); then
+        echo "Maximum number of attempts exceeded for bucket $BUCKET_PATH - please check the error message and try again"
+        echo ""
+        break
+      fi
+      let COUNTER=COUNTER+1
+      # retry with a project to bill in case it failed with a 400 error
+      PROJECT_TO_BILL="-u broad-dsde-prod"
```

Comment:
why do you set this again here? what if it was set to "" above? doesn't this change the meaning of the call when you retry?